src/components/AddComment.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import { postComment } from "../utils";
+import "../CSS/AddComment.css"
+
+const AddComment = ({article_id, setComments}) => {
+
+    const [newComment, setNewComment] = useState("");
+    const [commentPosted, setCommentPosted] = useState(false);
+
+    const handleNewComment = (e) => {
+        e.preventDefault();
+        postComment(article_id, newComment).then(({comment}) => {
+            setComments((currComments) => {
```

Comment:
Because the settting of state for the comments is in a then block, it's dependent on the post promise resolving. Invoke the setComments function outside of a promise chain. If you want to dot the I's and cross the T's then have a catch block on the end of postcomment that can handle the situation that the API rejects the promise.